tests/integration_tests/cmd/test_clean.py

```diff
@@ -32,92 +32,96 @@
 
 
 @pytest.mark.user_data(USER_DATA)
-class TestCleanCommand:
-    @pytest.mark.skipif(
-        not IS_UBUNTU, reason="Hasn't been tested on other distros"
+@pytest.mark.skipif(
+    not IS_UBUNTU, reason="Hasn't been tested on other distros"
+)
+def test_clean_rotated_logs(client: IntegrationInstance):
```

Comment:
Why are you making these tests separate functions? Per our [contribution guidelines](https://cloudinit.readthedocs.io/en/latest/development/testing.html):

> For ease of organisation and greater accessibility for developers unfamiliar with pytest, all cloud-init unit tests must be contained within test classes. In other words, module-level test functions should not be used.

Since you want the a an instance per test, shouldn't the `client` fixture suffice?

Reply:
You are correct @holman I conflated separate functions not the separate pytest fixtures with the solution. I thought both were needed, not just a change in fixture.


And sure enough per the docstring on the fixtures:
 - [class_client: used with reuse of same instance within the class is desired](https://github.com/canonical/cloud-init/blob/main/tests/integration_tests/conftest.py#L297)
 - [client: used with new instance launch per test is desired](https://github.com/canonical/cloud-init/blob/main/tests/integration_tests/conftest.py#L279)